### PR TITLE
Fix GC tests that contain GC.Collect() used in wrong scope

### PR DESCRIPTION
**Pull request by sywhang**
This same fix has been made in #17656 and #17594. I went through all the GC tests and fixed places where we observe a similar issue. Note that most of these tests don't necessarily depend on this fix, since the JIT won't inline methods / extend lifetime of temps for most of these tests, but wanted to fix these before we see more test failures coming from the various scenarios these tests will be running against.

### CODE REVIEW

File: tests/src/GC/Features/Finalizer/finalizeother/finalizedest.cs [hunk 1]
```diff
@@ -33,22 +34,22 @@ public CreateObj()
...
         }
     }
 
     public static int Main()
     {
         CreateObj temp = new CreateObj();
+        temp.RunTest();
+
+        GC.Collect(); 
+        GC.WaitForPendingFinalizers();  // makes sure Finalize() is called.
```
Comment (Maoni0) [inline tests/src/GC/Features/Finalizer/finalizeother/finalizedest.cs]: >WaitForPendingFinalizers [](start = 11, length = 24)

might as well add GC.Collect() here to be consistent

**Comment by sywhang** [comment]
By utility function, I assume you mean just for all the GC tests and not like adding a managed API right?

**Comment by Maoni0** [comment]
that's correct. it's a test utility function.

**Comment by sywhang** [comment]
Ok, I'll throw that in and change the tests accordingly.

**Pull request merged**